### ISSUE TITLE
Running `cargo test` causes a segfault.

### ISSUE THREAD

**Issue by tekul** [denoland/deno_core#217]
The details are as reported in https://github.com/denoland/deno/issues/19926. Essentially we are using deno embedded in a project and started seeing segfaults after a particular version. I noticed that I saw the same issue when running `cargo test` in deno, and also in deno_core (anything with v8 after version 0.74.3).

As described in my last comment in that issue, I get a segfault running `cargo clean` followed by `cargo test ` with the latest deno_core 6a369d5702be.

```
Thread 46 "modules::tests:" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff63f36c0 (LWP 36038)]
0x00005555569e23da in v8::internal::ThreadIsolation::RegisterJitPage(unsigned long, unsigned long) () at ../../../../v8/src/common/code-memory-access.cc:362
```

The machine is running Arch Linux:

```
$ uname -a
Linux celeste 6.5.3-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 13 Sep 2023 08:37:40 +0000 x86_64 GNU/Linux
```

I also get the same error on a different laptop using the same OS.

**Comment by lucacasonato** [comment]
This is a bug on Linux on modern Intel & AMD CPUs. We are aware of the issue, and are planning to address it soon. @tekul running individual tests should succeed - it's just running the entire suite that fails.

**Comment by juli1** [comment]
Also impacted by this bug that makes our software crash on Intel.

**Comment by juli1** [comment]
Bump this, this is impacting users and there is still no solution today.

**Comment by tekul** [comment]
A workaround we found if your project is using deno in rust tests is to use the `ctor` crate to make sure the runtime is initialized before the tests are run. i.e. in `lib.rs` add:

```rust
#[cfg(test)]
use ctor::ctor;

#[cfg(test)]
#[ctor]
// Make sure deno runtime is initialized in the main thread in test executables.
fn init_deno_runtime() {
    deno_core::JsRuntime::init_platform(None);
}
```

**Comment by Digifox03** [comment]
This was fixed by #471
